susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py

```diff
@@ -132,9 +130,18 @@ def _connect_db():
             log.error("Error on getting database pillar: %s", err.args)
             return
     try:
-        yield cursor
+        func(cursor)
     except psycopg2.DatabaseError as err:
-        log.error("Error on getting database pillar: %s", err.args)
+        log.error("Error on getting database pillar, trying again: %s", err.args)
```

Comment:
instead of implementing in this way, should it be put in a loop with the number of retries (in this case 2) and a break as soon we are able to get the func?

Reply:
ok, I changed the code